dcpy/test/models/product/test_metadata.py

```diff
@@ -122,3 +122,40 @@ def test_query_product_dataset_tags(test_metadata_repo: Path):
             product="lion", dataset="school_districts", destination="socrata"
         )
     ] == repo.query_dataset_destinations(TAG)
+
+
+@pytest.fixture
+def test_metadata_repo_snippets(resources_path: Path):
+    return resources_path / "test_product_metadata_repo_with_snippets"
+
+
+@pytest.fixture
+def dataset_with_snippets(test_metadata_repo_snippets: Path):
+    repo = md.OrgMetadata.from_path(
+        test_metadata_repo_snippets, template_vars={"version": "VERSION"}
+    )
+    product = repo.product("test_product")
+    return product.get_product_dataset("test_dataset", product.get_product_metadata())
```

Comment:
This pattern of passing around the full Metadata is one that I introduced, and I wince every time I see it. Maybe an unfair ask, but is it possible to tweak things in the OrgMetadata and ProductFolder so that we can just do 
```
product = repo.product("test_product").dataset("test_dataset") 
```
and have that dataset be fully snippeted?

Reply:
See https://github.com/NYCPlanning/data-engineering/pull/1162/commits/7280c5aab84c676acc3af85a3da047781245a713